src/addon-loader.js

```diff
@@ -21,56 +21,17 @@ const PluginClient = require('./plugin/plugin-client');
 const db = require('./db');
 const Settings = require('./models/settings');
 const sleep = require('./sleep');
-const fs = require('fs');
 const path = require('path');
 
 // Open the database.
 db.open();
 
 async function loadAddon(addonPath, verbose) {
-  // Skip if there's no package.json file.
-  const packageJson = path.join(addonPath, 'package.json');
-  if (!fs.lstatSync(packageJson).isFile()) {
-    const err = `package.json not found: ${packageJson}`;
-    return Promise.reject(err);
-  }
-
-  // Read the package.json file.
-  let data;
-  try {
-    data = fs.readFileSync(packageJson);
-  } catch (e) {
-    const err =
-      `Failed to read package.json: ${packageJson}\n${e}`;
-    return Promise.reject(err);
-  }
-
-  let manifest;
-  try {
-    manifest = JSON.parse(data);
-  } catch (e) {
-    const err =
-      `Failed to parse package.json: ${packageJson}\n${e}`;
-    return Promise.reject(err);
-  }
-
-  // Verify API version.
-  const apiVersion = config.get('addonManager.api');
-  if (manifest.moziot.api.min > apiVersion ||
-      manifest.moziot.api.max < apiVersion) {
-    console.error(
-      `API mismatch for package: ${manifest.name}\n` +
-      `Current: ${apiVersion} ` +
-      `Supported: ${manifest.moziot.api.min}-${manifest.moziot.api.max}`);
-
-    const err =
-      `Failed to start ${manifest.display_name} add-on: API version mismatch`;
-    return Promise.reject(err);
-  }
+  const packageName = path.basename(addonPath);
```

Comment:
I see you factored the above into addon-utils. Shouldn't there be a call to validate the manifest?

Reply:
My thought was this:
1. The add-on loader wasn't really validating anything but the API version. It was primarily reading the manifest in order to get the name and display name.
2. Add-on manager is what ends up creating the addon-loader process.
3. Add-on manager already does full validation.

So, it seemed unnecessary to validate the manifest twice.

Reply:
Seems fair. I think that the important validation is the one done by the addon manager.